### ISSUE TITLE
Please list filenames of images with corrupt EXIF data

### ISSUE THREAD

**Issue by LaurentBonnaud** [saimn/sigal#394]
Hi,

when I run sigal on my image gallery, I see the following error message:

```
Collecting albums //usr/lib/python3/dist-packages/PIL/TiffImagePlugin.py:788: UserWarning: Corrupt EXIF data.  Expecting to read 12 bytes but only got 8.
  warnings.warn(str(msg))
Collecting albums, done.
```

I would like to inspect/fix the affected images, but unfortunately sigal does not display their filenames.

**Comment by saimn** [comment]
Hi,
The warning comes from Pillow, but if you run sigal in verbose mode (`-v`) it would print the filenames.

**Comment by LaurentBonnaud** [comment]
Unfortunately, it does not:
```
$ sigal build --verbose
INFO: Reading settings ...
INFO:       Source : /var/www/html/photos-export-sigal/pictures
INFO:  Destination : /var/www/html/photos-export-sigal/_build
INFO:        Theme : photoswipe
INFO: Using 2 cores
/usr/lib/python3/dist-packages/PIL/TiffImagePlugin.py:788: UserWarning: Corrupt EXIF data.  Expecting to read 12 bytes but only got 8.
  warnings.warn(str(msg))
INFO: Skip empty album: <Album>(path='2010-2011', title='2010-2011')
[...]
```

**Comment by saimn** [comment]
Hmm, maybe in debug mode then: `sigal build -d -n 1`

**Comment by LaurentBonnaud** [comment]
Thanks for the suggestions!

Unfortunately debug mode still does not display filenames:
```
$ sigal build -d -n 1
INFO - Reading settings ...
DEBUG - Settings file: sigal.conf.py
DEBUG - Rewrite source : pictures -> /var/www/html/photos-export-sigal/pictures
DEBUG - Rewrite destination : _build -> /var/www/html/photos-export-sigal/_build
DEBUG - Settings:
[...]
INFO -       Source : /var/www/html/photos-export-sigal/pictures
INFO -  Destination : /var/www/html/photos-export-sigal/_build
INFO -        Theme : photoswipe
DEBUG - Plugin paths: []
DEBUG - Registered plugin sigal.plugins.extended_caching
INFO - Using 1 cores
DEBUG - Loaded cache with 74795 entries
/usr/lib/python3/dist-packages/PIL/TiffImagePlugin.py:788: UserWarning: Corrupt EXIF data.  Expecting to read 12 bytes but only got 8.
  warnings.warn(str(msg))
INFO - Skip empty album: <Album>(path='2010-2011', title='2010-2011')
```

**Comment by saimn** [comment]
Annoying! We already catch some warnings from Pillow, so I guess we could just convert all warnings to logging messages and add the file name.

https://github.com/saimn/sigal/blob/964c27f9464f7bf8b2573bb7bfced3e6101229f3/sigal/image.py#L65-L81

**Comment by saimn** [comment]
@LaurentBonnaud - Just pushed a commit to convert all pillow warnings to logging message with the filename. If you can give a try, would be great!

**Comment by LaurentBonnaud** [comment]
Thanks for the patch!
I am currently on holiday and will test this as soon as I am back...

**Comment by LaurentBonnaud** [comment]
I upgraded sigal to version 2.1.1, but unfortunately I am still getting the PIL warning without a filename (the output did not change).

**Comment by saimn** [comment]
Hum, maybe it happens only when reading EXIF ... could you with the exif-warnings branch ? (or applying the patch locally, https://github.com/saimn/sigal/tree/exif-warnings ; the commit is https://github.com/saimn/sigal/commit/fbdac965a9e3fda0a490db626d4e2f58f285ff10)

**Comment by LaurentBonnaud** [comment]
Thanks a lot for this new patch!

The warning is now caught, but there is still no filename:
```
Reading settings ...
Source : /var/www/html/photos-export-sigal/pictures
Destination : /var/www/html/photos-export-sigal/_build
Theme : photoswipe
Using 2 cores
PILImage reported a warning for file <PIL.MpoImagePlugin.MpoImageFile image mode=RGB size=4896x2752 at 0x7FEFE52F5790>
<class 'UserWarning'>: Corrupt EXIF data.  Expecting to read 12 bytes but only got 8.
```

**Comment by saimn** [comment]
Indeed, fixed in 5f0929cf0e075a9b8d1fe17745d9b9b3b97c7d6c and this is now in master.
Thanks for the help to improve those warnings!

**Comment by LaurentBonnaud** [comment]
> Indeed, fixed in 5f0929c and this is now in master.

I tested this new version and I could locate the affected file :>.

> Thanks for the help to improve those warnings!

Thanks for the fix!
